lisp/mastodon-tl.el

```diff
@@ -356,14 +525,23 @@ UPDATE-FUNCTION is used to recieve more toots."
          (json (mastodon-http--get-json url)))
     (with-output-to-temp-buffer buffer
       (switch-to-buffer buffer)
+      (setq
+       ;; Initialize with a minimal interval; we re-scan at least once
+       ;; every 5 minutes to catch any timestamps we may have missed
+       mastodon-tl--timestamp-next-update (time-add (current-time)
+                                                    (seconds-to-time 300)))
       (funcall update-function json))
     (mastodon-mode)
     (with-current-buffer buffer
-      (make-local-variable 'mastodon-tl--buffer-spec)
       (setq mastodon-tl--buffer-spec
             `(buffer-name ,buffer-name
                           endpoint ,endpoint update-function
-                          ,update-function)))
+                          ,update-function)
+            mastodon-tl--timestamp-update-timer (run-at-time mastodon-tl--timestamp-next-update
+                                                             nil ;; don't repeat
```

Comment:
I added a customizable var to switch it all off.

Note that the 5mins mentioned above is the upper limit for a refresh; we could up that now (I'm fairly confident my next-update calculation is working properly now)...